File: December - 17/KishoreMuruganantham_racing_turtles.R
input <- file("stdin", "r")
lines <- readLines(input)

p_val <- function(l) {
  l <- sub(".*=\\s*", "", l)
  l <- gsub("\\[|\\]", "", l)
  as.numeric(unlist(strsplit(l, ",")))
}

target <- p_val(lines[1])
n <- p_val(lines[2])

if (n == 0) {
  cat("No turtle fleets formed.\n")
} else {
  pos <- p_val(lines[3])
  spd <- p_val(lines[4])
  
  times <- (target - pos) / spd
  
  ord <- order(pos, decreasing = TRUE)
  sorted_times <- times[ord]
  
  fleets <- 0
  current_max <- 0
  
  for (t in sorted_times) {
    if (t > current_max) {
      fleets <- fleets + 1
      current_max <- t
    }
  }
  
  cat("The number of turtle fleets is: ", fleets, "\n", sep = "")
}

File: December - 17/KishoreMuruganantham_racing_turtles.swift
import Foundation

var lines = [String]()
while let line = readLine() { lines.append(line) }

func parse(_ l: String) -> [Double] {
    var s = l
    if let r = s.range(of: "=") { s = String(s[r.upperBound...]) }
    s = s.replacingOccurrences(of: "[", with: "")
    s = s.replacingOccurrences(of: "]", with: "")
    return s.split(separator: ",").compactMap { Double($0.trimmingCharacters(in: .whitespaces)) }
}

if lines.count >= 4 {
    let target = parse(lines[0]).first ?? 0
    let n = Int(parse(lines[1]).first ?? 0)
    
    if n == 0 {
        print("No turtle fleets formed.")
    } else {
        let pos = parse(lines[2])
        let spd = parse(lines[3])
        
        var turtles = [(Double, Double)]()
        for i in 0..<n {
            if i < pos.count && i < spd.count {
                turtles.append((pos[i], spd[i]))
            }
        }
        
        turtles.sort { $0.0 > $1.0 }
        
        var times = [Double]()
        for t in turtles {
             times.append((target - t.0) / t.1)
        }
        
        var fleets = 0
        var currentMax = 0.0
        
        for time in times {
            if time > currentMax {
                fleets += 1
                currentMax = time
            }
        }
        print("The number of turtle fleets is: \(fleets)")
    }
}
